mathjax3-ts/input/tex/autoload/AutoloadConfiguration.ts

```diff
@@ -0,0 +1,140 @@
+ *
+ * @param {TexParser} parser   The TeX input parser
+ * @param {string} name        The control sequence that is running
+ * @param {string} extension   The extension to load
+ */
+function Autoload(parser: TexParser, name: string, extension: string) {
+    if (Package.packages.has(parser.options.require.prefix + extension)) {
+        for (const macro of parser.options.autoload[extension]) {
+            AutoloadMap.remove(macro);
+        }
+        parser.i -= name.length;  // back up and read the macro again
+    }
+    RequireLoad(parser, extension);
+}
+
+/**
+ * Check if the require extension has been initialized
+ * (If autoload has been included in the TeX packages, but require isn't, then we need
+ *  to set up the options here and configure the require package in configAutoload below.
+ *  the priorities of the initialization and configuration are set so that autoload
+ *  will run after require when both are used.)
+ */
+function initAutoload(config: Configuration) {
+    if (!config.options.require) {
+        defaultOptions(config.options, RequireConfiguration.options);
+    }
+}
+
+/**
+ * Create the macros for the extensions that need to be loaded.
+ * Only ones that aren't already defined are made to autoload
+ *   (except for \color, which is overridden if present)
+ */
+function configAutoload(config: Configuration, jax: TeX<any, any, any>) {
+    const parser = jax.parseOptions;
+    const macros = parser.handlers.get('macro');
+    const autoload = parser.options.autoload;
+    for (const extension of Object.keys(autoload)) {
+        for (const name of autoload[extension]) {
+            if (!macros.lookup(name) || name === 'color') {
+                AutoloadMap.add(name, new Macro(name, Autoload, [extension]));
+            }
+        }
+    }
+    //
+    //  Check if the require extension needs to be configured
+    //
+    if (!parser.options.require.jax) {
+        RequireConfiguration.config(config, jax);
+    }
+}
+
+/**
+ * The command map for the macros that autoload extensions
+ */
+const AutoloadMap = new AutoloadCommandMap('autoload', {}, {});
+
+/**
+ * The configuration object for configMacros
+ */
+export const AutoloadConfiguration = Configuration.create(
+    'autoload', {
+        handler: {macro: ['autoload']},
+        options: {
+            //
+            //  These are the extension names and the macros they contains.
+            //  You can prevent one from being autoloaded by setting
+            //    it to [] in the options when the TeX input jax is created.
+            //  You can include the prefix if it is not the default one from require
+            //
+            autoload: expandable({
+                action: ['toggle', 'mathtip', 'texttip'],
+                bbox: ['bbox'],
+                boldsymbol: ['boldsymbol'],
+                braket: ['bra', 'ket', 'braket', 'set', 'Bra', 'Ket', 'Braket', 'Set', 'ketbra', 'Ketbra'],
```

Comment:
This will clash with the physics package.

Reply:
If `physics` is loaded, either by `\require` or through including it in the configuration of the packages to be loaded, then its versions of these macros will supersede the ones from `autoload`.  If `bracket` has already been autoloader and `\require{physics}` is executed, the physics versions will supersede the `bracket` ones.